### PR TITLE
[PECO-1706] Create separate projects for PySQL and SQLAlchemy Dialect Split

### PR DESCRIPTION
**Pull request by jprakash-db**
Project structure changed to have two separate projects for pysql and sqlalchemy-dialect.
Related tickets:
https://databricks.atlassian.net/browse/PECO-1706

Design Doc:
https://docs.google.com/document/d/1B0__xLR6r1p0qYXyQeHUtJbkkRGenwRknc_h4VuFwBo/edit

### CODE REVIEW

File: test_split.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import pandas as pd
```
Comment (kravets-levko) [inline test_split.py]: I think this file is redundant

**Review by kravets-levko — COMMENTED** [review]
I'm pleasantly impressed by how it turned out, thanks for the great job! Few notes before we can move forward:

- let's use `databricks-sqlalchemy-dialect` name for SQLAlchemy plugin everywhere (in `pysql-connector/databricks/sqlalchemy/__init__.py`, and also rename the folder from `sqlalchemy-plugin` to `sqlalchemy-dialect`)
- I think, no need to have a `sqlalchemy_db` folder inside of `sqlalchemy-plugin/src` - just move all the files one level up

**Comment by kravets-levko** [comment]
Also, please take updates from upstream. And general advice - keep your fork up to date. We continue making changes in the main repo, and having both in sync will help to avoid merge conflicts in future

**Comment by gopalldb** [comment]
> Also, please take updates from upstream. And general advice - keep your fork up to date. We continue making changes in the main repo, and having both in sync will help to avoid merge conflicts in future

@kravets-levko Shouldn't we push to main repo for these changes?

**Comment by kravets-levko** [comment]
> Shouldn't we push to main repo for these changes?

No, Databricks for should remain stable until we're fully done. But to help ourselves in future - you should get updates from Databricks fork (its `main` branch) to keep your branches up to date and avoid conflicts when it will be time to bring all the changes back to Databricks repo

File: sqlalchemy-dialect/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+[tool.poetry]
+name = "databricks-sqlalchemy-dialect"
+version = "1.0.0"
+description = "SQLAlchemy dialect for integration with Databricks databases"
+authors = ["Databricks <databricks-sql-connector-maintainers@databricks.com>"]
+license = "Apache-2.0"
+readme = "README.md"
+packages = [{ include = "databricks_sqlalchemy_dialect" }]
```
Comment (kravets-levko) [inline sqlalchemy-dialect/pyproject.toml:8]: We use dashes for multi-word folder names, so `databricks-sqlalchemy-dialect`, not `databricks_sqlalchemy_dialect`, so please rename. After that we should be good to go

Comment (shivam2680) [reply sqlalchemy-dialect/pyproject.toml:8]: If we use dashes, then we won't be able to import in the code right? 
`from databricks-sqlalchemy-dialect` gives error.

**Pull request merged**